pypesto/objective/aggregate.py

```diff
@@ -0,0 +1,175 @@
+        """
+        Constructor.
+
+        Parameters
+        ----------
+
+        objectives: list
+            List of pypesto.objetive instances
+
+        """
+        # input typechecks
+        if not isinstance(objectives, list):
+            raise TypeError(f'Objectives must be a list, '
+                            f'was {type(objectives)}.')
+
+        if not all(
+                isinstance(objective, Objective)
+                for objective in objectives
+        ):
+            raise TypeError('Objectives must only contain elements of type'
+                            'pypesto.Objective')
+
+        self.objectives = objectives
+
+        # assemble a dict that we can pass as kwargs to the
+        # pypesto.Objective constructor
+        init_kwargs = {
+            'x_names': x_names,
+            'options': options
+        }
+
+        # check if all objectives consistently accept sensi orders in fun/res
+        # and adopt the same behaviour in aggregate
+        for attr in ['fun_accept_sensi_orders', 'res_accept_sensi_orders']:
+            _check_boolean_value_consistent(objectives, attr)
+            init_kwargs[attr] = getattr(objectives[0], attr)
+
+        # check if all objectives consistently accept sensi orders in fun/res
+        # and adopt the same behaviour in aggregate
+        for attr in [
+            'fun', 'grad', 'hess', 'hessp', 'res', 'sres'
+        ]:
+            if any(
+                getattr(objective, attr) is None
+                for objective in objectives
+            ):
+                init_kwargs[attr] = None
+            elif all(
+                isinstance(getattr(objective, attr), bool)
+                for objective in objectives
+            ):
+                _check_boolean_value_consistent(objectives, attr)
+                init_kwargs[attr] = getattr(objectives[0], attr)
+            elif all(
+                callable(getattr(objective, attr))
+                for objective in objectives
+            ):
+                aggregate_fun = f'aggregate_{attr}'
+                if (
+                        attr == 'fun'
+                        and init_kwargs['fun_accept_sensi_orders']
+                ) or (
+                        attr == 'res'
+                        and init_kwargs['res_accept_sensi_orders']
+                ):
+                    aggregate_fun += '_sensi_orders'
+
+                init_kwargs[attr] = getattr(self, aggregate_fun)
+            else:
+                raise ValueError(f'{attr} has incompatible types across '
+                                 f'instances!')
+
+        super().__init__(**init_kwargs)
+
+    def aggregate_fun_sensi_orders(self, x, sensi_orders):
+        rvals = [
+            objective.fun(x, sensi_orders)
+            for objective in self.objectives
+        ]
+        return {
+            key: sum(rval[key] for rval in rvals)
+            for key in ['fval', 'grad', 'hess']
+            if key in rvals[0]
+        }
```

Comment:
one issue: it is possible to call an objective with ``return_dict = True``. In that case the whole dict that is created is returned. This was introduced to have an easy way to get the ``RDATAS`` from AmiciObjectives. Problem now is that these get lost in the aggregation step. Maybe one could check whether an RDATAS field exists (pypesto.constants.RDATAS), and then just concatenate them here?